test/mmapper/mmapper.c

```diff
@@ -0,0 +1,93 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sched.h>
+#include <signal.h>
+#include <string.h>
+
+#ifndef CLONE_NEWPID
+#define CLONE_NEWPID 0x20000000
+#endif
+
+static int do_test(void *logf)
+{
+    int fd, i = 0;
+
+    // Detach from the parent terminal
+    setsid();
+
+    // Close standard file descriptors
+    close(STDIN_FILENO);
+    close(STDOUT_FILENO);
+    close(STDERR_FILENO);
+
+    // Open /dev/null as stdin
+    fd = open("/dev/null", O_RDONLY);
+    if (fd != STDIN_FILENO)
+    {
+        dup2(fd, STDIN_FILENO);
+        close(fd);
+    }
+
+    // Open the log file and redirect stdout and stderr to it
+    fd = open(logf, O_WRONLY | O_TRUNC | O_CREAT, 0600);
+    dup2(fd, STDOUT_FILENO);
+    dup2(fd, STDERR_FILENO);
+    if (fd != STDOUT_FILENO && fd != STDERR_FILENO)
+        close(fd);
+
+    while (1)
+    {
+        sleep(1);
+        printf("%d\n", i++);
+        fflush(stdout);
+    }
+
+    return 0;
+}
+
+int main(int argc, char *argv[])
+{
+    if (argc < 3)
+    {
+        fprintf(stderr, "Usage: %s <size> <mapping_flags>\n", argv[0]);
+        return 1;
+    }
+
+    int pid;
+    void *stk;
+
+    size_t size = atoi(argv[1]);
+    int flags = MAP_ANON;
+
+    if (strcmp(argv[2], "MAP_SHARED") == 0)
```

Comment:
Should we use `s` and `p` instead of `MAP_SHARED` and `MAP_PRIVATE`, and perhaps `sa` for `MAP_SHARED|MAP_ANONYMOUS`?